htlcswitch/hop/iterator.go

```diff
@@ -864,11 +867,12 @@ func (p *OnionProcessor) DecodeHopIterators(id []byte,
 			continue
 		}
 
-		// If this index is contained in the replay set, mark it with a
-		// temporary channel failure error code. We infer that the
-		// offending error was due to a replayed packet because this
-		// index was found in the replay set.
-		if replays.Contains(uint16(i)) {
+		// If this index is contained in the replay set, and it is not a
+		// reforwarding on startup, mark it with a temporary channel
+		// failure error code. We infer that the offending error was due
+		// to a replayed packet because this index was found in the
+		// replay set.
```

Comment:
could you add a `NOTE` here, that in the case of a `reforward` all the previous HTLCs are part of the replayed log, and therefore `reforwards` would falsely flagged as replays without this important check.

Reply:
hmmm how can reforwarded htlcs be flagged as replays? or are you saying without this check, reforwards will be considered as replays? isn't that's what the above comment is saying?

Reply:
Let's keep it that way if you think it already describes the circumstances however please change the temporary channel failure  => to permanent see BOLT 04 errors:

```
type: BADONION|PERM|4 (invalid_onion_version)
data:
[sha256:sha256_of_onion]
```

it sets the PERM bit.

Reply:
I think we are expanding the scope again, and, why this `invalid_onion_version` error?

Reply:
I don't think we change the scope here much, we send back the error `invalid_onion_version` (legacy code) and what I am proposing is that we describe correctly that this is not a temporary error but a permanent one ? 

https://github.com/lightningnetwork/lnd/blob/35102e7c35c7923e27d937d10ba51757edd9ec6e/htlcswitch/hop/iterator.go#L881

Reply:
oh i see you meant the comments😂 Thought you wanna use a new error code or sth.